src/lib/common/JsonHelper.cpp

```diff
@@ -0,0 +1,153 @@
+/*
+*
+* Copyright 2013 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms
+* of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Orion dev team
+*/
+
+#include "common/JsonHelper.h"
+
+/* ****************************************************************************
+*
+* toJsonString -
+*/
+std::string toJsonString(const std::string& input)
+{
+  std::ostringstream ss;
+
+  ss << '"';
+  for (std::string::const_iterator iter = input.begin(); iter != input.end(); ++iter)
+  {
+    switch (char ch = *iter)
+    {
+    case '\\': ss << "\\\\"; break;
+    case '"': ss << "\\\""; break;
+    case '/': ss << "\\/"; break;
+    case '\b': ss << "\\b"; break;
+    case '\f': ss << "\\f"; break;
+    case '\n': ss << "\\n"; break;
+    case '\r': ss << "\\r"; break;
+    case '\t': ss << "\\t"; break;
+    default:
+      if (ch >= 0 && ch <= 0x1F)
+      {
+        static const char intToHex[16] =  { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F' } ;
+
+        ss << "\\u00" << intToHex[(ch & 0xF0) >> 4] << intToHex[ch & 0x0F];
```

Comment:
Took me awhile to understand this ...
It's a 'hex2acsii', like a printf("%x", ch), covering all special chars before SPACE (32)
I would add a comment - something like "Converting the rest of special chars 0-31 to \u00xx".
Perhaps there is even a name for \u00xx?

What about chars 0x80 - 0xFF - should they also be covered by this conversion?